docs_guide/apidocs/how_to_add_an_api_reference.rst

```diff
@@ -8,13 +8,124 @@ This guide shows how to add an API reference page to your documentation.
 
 .. include:: ../others/general_pre-requisites.rst
 
-Create a new API reference
-==========================
+Set up a new project API reference
+==================================
 
-In order to create an API reference, you need to find the function or class for which you want to
-write the reference in the main code of your project. In Qiskit, we are following `the Google Python
-Style Guide <https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings>`_ for
-docstrings. 
+If your project doesn't have a :doc:`Sphinx <sphinx:index>`-based API reference yet, you need to add ``"sphinx.ext.autodoc"``,
+``"sphinx.ext.autosummary"`` and ``"sphinx.ext.napoleon"`` to the :confval:`sphinx:extensions` variable of ``docs/conf.py``.
+
+Also in ``docs/conf.py``, it is recommended to set the following configuration variables:
+
+* :confval:`sphinx:autosummary_generate` to ``True`` to make sure pages are scanned for autosummary directives.
+* :confval:`sphinx:autosummary_generate_overwrite` to ``False`` so that :mod:`sphinx:sphinx.ext.autosummary` doesn't overwrite existing files by generated stub files.
+* :confval:`sphinx:autodoc_typehints` to ``"description"`` so the type hints are only shown as part of the function or method instead of in the signature.
+* :confval:`sphinx:autodoc_typehints_description_target` to ``"documented_params"`` so that only the types of parameters and return values that are included in the docstring.
+* :confval:`sphinx:autoclass_content` to ``"both"`` so when you use the :rst:dir:`sphinx:autoclass` directive you get the docstrings of both the class and the ``__init__()`` method.
```

Comment:
Although your text explanation is clear, but I think it's still better to nclude a code cell with these configurations? Something like

```
autosummary_generate = True
...
```

Reply:
Done in [e937d2d](https://github.com/Qiskit/qiskit_sphinx_theme/pull/284/commits/e937d2d68d95654804b79fc5aba9be6da2d42ac6)